metabase/metabase
# Fix row id mapping in object detail modal displaying

Closes https://github.com/metabase/metabase/issues/39477
Closes https://github.com/metabase/metabase/issues/34070

### Description

we have a not straightforward logic on detecting what the data should be shown after clicking on a table row to show object detail modal. This PR reuses logic of mapping row id to the entity id

### How to verify

- create a new question from orders table
- join people table
- click on visualize
- click on the most left cell of any displayed row
- click on first row should work
- click on any row should display a modal with the correct id (same id as the row you clicked has)

### Demo

Before

https://github.com/metabase/metabase/assets/125459446/adaff085-e940-40e4-8ade-b0eabb710214

After


https://github.com/metabase/metabase/assets/125459446/42d3a398-9278-4980-8687-6878699f8a33




### Checklist

- [x] stress test https://github.com/metabase/metabase/actions/runs/8527115010
